bolt/lib/Profile/YAMLProfileWriter.cpp

```diff
@@ -27,25 +28,55 @@ namespace bolt {
 
 /// Set CallSiteInfo destination fields from \p Symbol and return a target
 /// BinaryFunction for that symbol.
-static const BinaryFunction *setCSIDestination(const BinaryContext &BC,
-                                               yaml::bolt::CallSiteInfo &CSI,
-                                               const MCSymbol *Symbol) {
+static const BinaryFunction *
+setCSIDestination(const BinaryContext &BC, yaml::bolt::CallSiteInfo &CSI,
+                  const MCSymbol *Symbol, const BoltAddressTranslation *BAT) {
   CSI.DestId = 0; // designated for unknown functions
   CSI.EntryDiscriminator = 0;
+  auto setBATSecondaryEntry = [&](const BinaryFunction *const Callee) {
+    // The symbol could be a secondary entry in a cold fragment.
+    ErrorOr<uint64_t> SymbolValue = BC.getSymbolValue(*Symbol);
+    if (SymbolValue.getError())
```

Comment:
Can we get a condition when `getFunctionForSymbol(Symbol)` return a function, but the symbol has no set value?

Reply:
We shouldn't. Let me just use `cantFail` wrapper here.